File: src/apps/NotificationWindow/src/NotificationWindowView.m
#import "NotificationWindowView.h"

@interface NotificationWindowView ()

@property(weak) IBOutlet NSImageView* closeButton;

@end

@implementation NotificationWindowView

- (void)drawRect:(NSRect)dirtyRect {
  [NSGraphicsContext saveGraphicsState];
  {
    [NSColor.windowBackgroundColor set];
    [[NSBezierPath bezierPathWithRoundedRect:self.bounds xRadius:12 yRadius:12] fill];
  }
  [NSGraphicsContext restoreGraphicsState];
}

- (IBAction)hideWindow:(id)sender {
  [self.window orderOut:self];
}

@end

File: src/apps/NotificationWindow/src/NotificationWindowView.swift
public class NotificationWindowView: NSView {
    @IBOutlet var closeButton: NSImageView!
    @IBOutlet var text: NSTextField!

    override public func draw(_: CGRect) {
        NSGraphicsContext.saveGraphicsState()

        NSColor.windowBackgroundColor.set()
        NSBezierPath(roundedRect: bounds, xRadius: 12, yRadius: 12).fill()

        NSGraphicsContext.restoreGraphicsState()
    }

    @IBAction func hideWindow(_: Any) {
        window?.orderOut(self)
    }
}
